internal/exec/stages/disks/partition_dasd.go

```diff
@@ -0,0 +1,298 @@
+	if mib != nil {
+		*mib = ((*mib) * 1024 * 1024) / (blocksPerTrack * sectorSize)
+	}
+}
+
+// recreateOrigParts recreates the original partition layout. This is used if partitions were only deleted, so the
+// remaining partitions are preserved. Note : preserved here still means the partitions will be recreated, but just that
+// the same layout will be maintained
+func recreateOrigParts(resultParts []types.Partition, diskInfo util.DiskInfo) []types.Partition {
+	excludeMap := map[int]bool{}
+	originalPartitions := []types.Partition{}
+	for _, part := range resultParts {
+		if partitionShouldExist(part) {
+			return originalPartitions
+		}
+		excludeMap[part.Number] = true
+	}
+
+	// add the rest of the partitions
+	for _, origPart := range diskInfo.Partitions {
+		if _, ok := excludeMap[origPart.Number]; ok {
+			continue
+		}
+		startTrack := origPart.StartTrack
+		sizeInTracks := origPart.SizeInTracks
+		part := types.Partition{
+			Number:   origPart.Number,
+			StartMiB: &startTrack,
+			SizeMiB:  &sizeInTracks,
+		}
+		originalPartitions = append(originalPartitions, part)
+	}
+	return originalPartitions
+}
+
+// getDasdRealStartAndSize returns a map of partition numbers to a struct that contains what their real start
+// and end track should be. Because there is no `pretend` equivalent for fdasd, it becomes a little
+// cumbersome and ugly to correctly determine the partition layout in all cases. So, there are some differences
+// to how partitioning with DASDs will work:
+//       - partitions will always be recreated
+//       - there is no explicit deletion of partitions as the non interactive way to create partitions always recreates
+//         the table.
+//       - the old partition layout will be destroyed and only the new ones will be created. This is because
+//         the non interactive fdasd command doesn't really support a 'delete' operation.
+//       - "deletion" of partitions  will still work BUT if a partition is also specified to be created
+//          along with deletion, the old layout will not be preserved. the reason for this is that there
+//          are so many cases to cover and it becomes really messy to do it by hand.
+// It also converts everything to tracks so the StartMiB/SizeMiB will NOT be in MiB after this call
+func getDasdRealStartAndSize(dev types.Disk, diskInfo util.DiskInfo) ([]types.Partition, error) {
+	result := []types.Partition{}
+	nextTrack := 0
+	startTrack := 2 // DASD partitions start at 2 for CDL
+
+	//pass1 : fill any values available from existing parts and convert MiB to tracks
+	for _, part := range dev.Partitions {
+		convertMiBToTracks(part.SizeMiB, diskInfo.BlocksPerTrack, diskInfo.LogicalSectorSize)
+		convertMiBToTracks(part.StartMiB, diskInfo.BlocksPerTrack, diskInfo.LogicalSectorSize)
+
+		if info, exists := diskInfo.GetPartition(part.Number); exists {
+			if part.StartMiB == nil {
+				part.StartMiB = &info.StartTrack
```

Comment:
the `types.Partitions` json is directly manipulated here which is why i did not want to add any new vars to it.

I thought to keep the implementation simple and to avoid merging partitions manually, I will take a look again to see if it can be done comprehensively, because I do not want to miss any cases and have the user end up with half baked partitioning.

Reply:
added a new partition structure as part of fdasd and used that for the translation.